src/nemos/glm.py

```diff
@@ -84,12 +81,11 @@ class GLM(BaseRegressor):
     def __init__(
         self,
         observation_model: obs.Observations = obs.PoissonObservations(),
-        regularizer: reg.Regularizer = reg.UnRegularized("GradientDescent"),
```

Comment:
this may break the sklearn api for `BaseEstimator`. the `estimator.get_params()` will return only the observation model if the parameter is not in the init. If we want to cv the regularization we have to have it at the init

```python
In [3]: mdl = nmo.glm.GLM()
In [4]: mdl.get_params()
Out[4]: 
{'observation_model__inverse_link_function': <PjitFunction of <function jax.numpy.exp at 0x33068f380>>,
 'observation_model': <nemos.observation_models.PoissonObservations at 0x1120e84d0>}
```

Reply:
so you want to pass a `regularizer` arg here? even if it just going to the `super()` call?

Reply:
we need to be able to cross-validate over "regularizer.regularizer_strength" for sure, so the get_params needs to be aware that we have a regularizer parameter.

Reply:
PS and get_params is only aware of positional or keyword and keyword only arguments of the`__init__`

Reply:
should be fixed now!

Reply:
yes! however i don't see the`regularizer__regularizer_strenght` for Ridge for the same reason. ridge only accepts **kwargs now in the init
```
import nemos as nmo
mdl = nmo.glm.GLM(regularizer=nmo.regularizer.Ridge())
mdl.get_params()
Out[7]: 
{'observation_model__inverse_link_function': <PjitFunction of <function jax.numpy.exp at 0x31d003600>>,
 'observation_model': <nemos.observation_models.PoissonObservations at 0x3195834d0>,
 'regularizer': <nemos.regularizer.Ridge at 0x105a442d0>,
 'solver_kwargs': {},
 'solver_name': 'GradientDescent'}

 ```

Reply:
ahhh, I see...will fix